src/components/graph/util/model-functions.ts

```diff
@@ -0,0 +1,146 @@
+        id: node.id,
+        type: node.type,
+        position: { x: 0, y: 0 },
+        data: data,
+    };
+}
+
+// Return the first node of type nodeType and specific buildStatus
+// in the tree model
+export function getFirstNodeOfType(
+    elements: NetworkModificationNode | RootNode,
+    nodeType: string,
+    buildStatusList: string[] | undefined
+) {
+    return recursiveSearchFirstNodeOfType(
+        elements,
+        undefined, // first is Root node without parent node
+        nodeType,
+        buildStatusList
+    );
+}
+export function isNetworkModificationNode(n: NetworkModificationNode | RootNode): n is NetworkModificationNode {
+    return 'nodeBuildStatus' in n;
+}
+
+// Recursive search of a node of type and buildStatus specified
+export function recursiveSearchFirstNodeOfType(
+    elements: NetworkModificationNode | RootNode,
+    parentNodeUuid: UUID | undefined,
+    nodeType: string,
+    buildStatusList: string[] | undefined
+): CurrentTreeNode | null {
+    const globalBuildStatus = isNetworkModificationNode(elements)
+        ? elements.nodeBuildStatus?.globalBuildStatus
+        : undefined;
+
+    if (
+        elements.type === nodeType &&
+        globalBuildStatus !== undefined &&
+        (buildStatusList === undefined || buildStatusList.includes(globalBuildStatus))
+    ) {
+        return convertNodetoReactFlowModelNode(elements, parentNodeUuid);
+    }
+
+    for (const child of elements.children ?? []) {
+        const found = recursiveSearchFirstNodeOfType(child, elements.id, nodeType, buildStatusList);
+        if (found) {
+            return found;
+        }
+    }
+    return null;
+}
+
+export function isNodeReadOnly(node: CurrentTreeNode | null) {
+    if (node?.type === NodeType.ROOT) {
+        return true;
+    }
+    return node?.data?.readOnly ? true : false; // ternary operator because of potential undefined
+}
+
+export function isNodeBuilt(node: CurrentTreeNode | null) {
+    if (!node) {
+        return false;
+    }
+    if (node.type === NodeType.ROOT) {
+        return true;
+    }
+    return node.data?.globalBuildStatus?.startsWith('BUILT');
+}
+
+export function isSameNode(node1: CurrentTreeNode | null, node2: CurrentTreeNode | null) {
+    return node1?.id === node2?.id;
+}
+
+export function isNodeRenamed(node1: CurrentTreeNode | null, node2: CurrentTreeNode | null) {
+    if (!node1 || !node2) {
+        return false;
+    }
+    return isSameNode(node1, node2) && node1?.data?.label !== node2?.data?.label;
+}
+
+export function isNodeInNotificationList(node: any, notificationIdList: any) {
+    if (!node || !notificationIdList) {
+        return false;
+    }
+    return notificationIdList.includes(node.id);
+}
+
+export function isSameNodeAndBuilt(node1: CurrentTreeNode | null, node2: CurrentTreeNode | null) {
+    return isSameNode(node1, node2) && isNodeBuilt(node1);
+}
+
+export function getAllChildren(elements: NetworkModificationTreeModel | null, nodeId: string) {
```

Comment:
```suggestion
export function getAllChildren(elements: NetworkModificationTreeModel | null, nodeId: UUID) {
```
I was it creates errors in another file
Don't spend too much time fixing them, if it's too long we can keep "string" here, but it doesn't hurt to clean the code a little, that's the point of those PR

Reply:
done